src/phalanx/models/environments.py

```diff
@@ -235,6 +235,34 @@ class EnvironmentConfig(EnvironmentBaseConfig):
         ),
     )
 
+    control_system_app_namespace: str | None = None
```

Comment:
The default value for these need to be Pydantic fields. The Pydantic Field contains the default value, and docstring goes into the Field's `title` and `description` arguments. There is no docstring for Pydantic fields.

Reply:
Additionally, I think we should move all these control system fields into a new Pydantic model class and then add that new `ControlSystemConfig` class as a `control_system` attribute on the `EnvironmentConfig` as an optional field. I.e.,

```
class EnvironmentConfig(EnvironmentBaseConfig):

    control_system: ControlSystemConfig | None = None
```

That'll help keep the control system application's global configuration fairly contained and clearly optional from Kubernetes/Argo CD environment configuration.

Reply:
The tests kept failing when trying to add the control system config object in the requested location. I had to move it into the `EnvironmentBaseConfig` definition to get things to pass.

Reply:
Yup, that's good. The difference between the two is whether the schema is also defined for the per-environment values files or just the merged settings. You want it defined for both, so putting it in base is correct.

I haven't seen a new push to this PR branch with that change, just FYI in case you'd thought you'd pushed it.

Reply:
The change was more extensive than I realized and I've just pushed it now. The way the control system configuration parameters are documented in `environments/values.yaml`, I've lost the ability to specify project wide settings and now have to do everything on a site level which isn't ideal. If you look at what I've done, can you tell me a way to handle project wide settings?

Reply:
To be clear, they should all still be nested under `controlSystem`, which I think was Jonathan's point. But `controlSystem` can be specified in `values.yaml`.